### PR TITLE
Create cap_anndata_schema.md

### PR DESCRIPTION
**Pull request by mfutey**
- Added missing sections, e.g. [cellannotation_setname]--category_fullname, [cellannotation_setname]--category_cell_ontology_exists, etc
- Added "source," "required for publication on CAP" and "example" fields and values for each section
- Checked for typos and consistency
- Changed format from bullet points to tables
- Removed character limit for descriptions
- Added 200 character limit for titles
- Corrected .var text to indicate ENSEMBL terms are required not gene names
- Removed "programmy" language, e.g. ndarray

### CODE REVIEW

**Review by evanbiederstedt — APPROVED** [review]
LGTM, sorry again @mfutey

File: cap_anndata_schema.md [hunk 1]
```diff
@@ -0,0 +1,1577 @@
...
+</tbody></table>
+
+
+### Clustering
+
+Users may OPTIONALLY include a single field for clustering within AnnData files, or multiple fields denoting clustering, e.g. different clustering algorithms, multiple resolutions of clustering, etc.
+
+We therefore REQUIRE that clustering is clearly denoted within the AnnData file if it contains clustering fields.
+
+ScanPy has set an AnnData community standard of defining the `*.obs` value by the type of algorithm. e.g. the function `scanpy.tl.louvain` (documented [here](https://scanpy.readthedocs.io/en/stable/generated/scanpy.tl.louvain.html)) by default saves the clustering as `anndata.obs['louvain']`. Similarly, `leiden` (documented [here](https://scanpy.readthedocs.io/en/stable/generated/scanpy.tl.leiden.html)) is often encoded as `anndata.obs['leiden']`.
```
Comment (rm1113) [inline cap_anndata_schema.md:344]: Please also note that we support two more key names: `leiden` and `louvian`. It means that user could use these names without `cluster_` prefix.

File: cap_anndata_schema.md [hunk 2]
```diff
@@ -0,0 +1,1577 @@
...
+  		<td><b>required for publication on CAP</b></td>
+  		<td>yes</td>
+	</tr>
+	<tr>
+  		<td><b>example</b></td>
+  		<td><code>'0.1.0'</code></td>
+	</tr>
+</tbody></table>
+
+## cap_dataset_title
```
Comment (rm1113) [inline cap_anndata_schema.md:1057]: In the previous version it was just `dataset_title`. I prefer more simple `title` to allign with CxG. What is the best name here? CC @evanbiederstedt @mfutey

File: cap_anndata_schema.md [hunk 3]
```diff
@@ -0,0 +1,1577 @@
...
+  		<td>yes</td>
+	</tr>
+	<tr>
+  		<td><b>example</b></td>
+  		<td><code>'0.1.0'</code></td>
+	</tr>
+</tbody></table>
+
+
+## cap_dataset_timestamp
```
Comment (rm1113) [inline cap_anndata_schema.md:996]: previosly was `cap_publication_timestamp`

File: cap_anndata_schema.md [hunk 4]
```diff
@@ -0,0 +1,1577 @@
...
+  		<td><b>required for publication on CAP</b></td>
+  		<td>yes</td>
+	</tr>
+	<tr>
+  		<td><b>example</b></td>
+  		<td><code>'2023-11-21T04:12:36'</code></td>
+	</tr>
+</tbody></table>
+
+## cap_dataset_version
```
Comment (rm1113) [inline cap_anndata_schema.md:1027]: was `cap_publication_version`

File: cap_anndata_schema.md [hunk 5]
```diff
@@ -0,0 +1,1577 @@
...
+  		<td>yes</td>
+	</tr>
+	<tr>
+  		<td><b>example</b></td>
+  		<td><code>'Human retina cell atlas - retinal ganglion cells'</code></td>
+	</tr>
+</tbody></table>
+
+
+## cap_dataset_description
```
Comment (rm1113) [inline cap_anndata_schema.md:1089]: was `dataset_description`

File: cap_anndata_schema.md [hunk 6]
```diff
@@ -0,0 +1,1577 @@
...
+  		<td><b>required for publication on CAP</b></td>
+  		<td>yes</td>
+	</tr>
+	<tr>
+  		<td><b>example</b></td>
+  		<td><code>'https://celltype.info/CAP_DataCuration/A-Single-Cell-Transcriptome-Atlas-of-the-Human-Pancreas/1/dataset/20'</code></td>
+	</tr>
+</tbody></table>
+
+## author_name
```
Comment (rm1113) [inline cap_anndata_schema.md:1244]: was `cap_author_name`

File: cap_anndata_schema.md [hunk 7]
```diff
@@ -0,0 +1,1577 @@
...
+  		<td>yes</td>
+	</tr>
+	<tr>
+  		<td><b>example</b></td>
+  		<td><code>'John Smith'</code></td>
+	</tr>
+</tbody></table>
+
+
+## author_contact
```
Comment (rm1113) [inline cap_anndata_schema.md:1276]: was `cap_author_contact`

File: cap_anndata_schema.md [hunk 8]
```diff
@@ -0,0 +1,1577 @@
...
+  		<td><b>required for publication on CAP</b></td>
+  		<td>yes</td>
+	</tr>
+	<tr>
+  		<td><b>example</b></td>
+  		<td><code>'jsmith@university.edu'</code></td>
+	</tr>
+</tbody></table>
+
+## orcid
```
Comment (rm1113) [inline cap_anndata_schema.md:1307]: was `cap_author_orcid`

File: cap_anndata_schema.md [hunk 9]
```diff
@@ -0,0 +1,1577 @@
...
+  		<td><b>required for publication on CAP</b></td>
+  		<td>yes</td>
+	</tr>
+	<tr>
+  		<td><b>example</b></td>
+  		<td><code>'{[cellannotation_set]--metadata:{'annotation_method':'algorithmic'...}}'</code></td>
+	</tr>
+</tbody></table>
+
+#### description
```
Comment (rm1113) [inline cap_anndata_schema.md:1373]: was `cellannotation_setdescription`. But I am happy with this change

**Review by rm1113 — CHANGES_REQUESTED** [review]
Please add the information about Clustering key names. Let's discuss the naming mismatch as well

File: cap_anndata_schema.md (hunk 2 above)
Comment (mfutey) [reply cap_anndata_schema.md]: I updated the field names to match those of the private repo, but we should update both versions to have more consistent names, e.g. `'dataset_title'`, `'dataset_description'` and `'dataset_url'` do not have the `cap_` prefix as the other fields at the dataset level.

**Review by rm1113 — APPROVED** [review]

File: cap_anndata_schema.md (hunk 2 above)
Comment (evanbiederstedt) [reply cap_anndata_schema.md]: Agreed, this is something @dosumis has been pushing as well.

File: cap_anndata_schema.md (hunk 1 above)
Comment (evanbiederstedt) [reply cap_anndata_schema.md:344]: I go back and forth on this issue. 

If `leiden` and `louvain` exist, they should be the only clustering in the file. 
Use case: the user then adds another level of clustering to a file with only `leiden`. Won't we then ask the user to format the clustering appropriately?

This makes me think that we should force users to format the clustering in a certain way.

Thoughts? @mfutey @unbelivr @rm1113 ?

Comment (mfutey) [reply cap_anndata_schema.md:344]: @rm1113 @evanbiederstedt I think we should always have the `'cluster'` prefix to keep the naming structure the same across datasets. 

Also the explanation of the column naming is already quite long, do we want to add another section stating you don't have to follow the three part naming convention of <code>'cluster + _ + [ALGORITHM_TYPE] + _ + [SUFFIX]'</code> ?

**Pull request merged**